docs/deploy/scrape.py

```diff
@@ -229,15 +232,37 @@ def upload(app_id, api_key, records, version):
     # Pick some settings for this index.
     index.set_settings(SETTINGS)
 
-    # Upload to temp index.
-    print(f"uploading {len(records)} records to temp index {temp_name}...", file=sys.stderr)
-    index.save_objects(records, {"autoGenerateObjectIDIfNotExist": True})
-    print("upload complete", file=sys.stderr)
-
-    # Rename index into place.
-    print(f"renaming temp index {temp_name} -> {final_name}...", file=sys.stderr)
-    client.move_index(temp_name, final_name).wait()
-    print("rename done", file=sys.stderr)
+    num_tries = 1
+    while num_tries <= max_retries:
```

Comment:
Use a `for`...`else` statement, the `else` fires if you don't `break` out of the `for` loop

```python
for try in range(max_retries):
  ...
  if success_criteria_are_met:
    break
  print(f"success criteria not met, got {result} instead")
else:
  raise ValueError(f"{max_retries} failures, giving up")
```